File: WrecklessEngine/Mouse.cpp
#include "Mouse.h"
#include <Windows.h>

namespace Input
{

	void Mouse::SetPosition(int x, int y)
	{
		SetCursorPos(x, y);
	}

	int Mouse::GetPositionX()
	{
		return GetPosition().X;
	}

	int Mouse::GetPositionY()
	{
		return GetPosition().Y;
	}

	MouseCoords Mouse::GetPosition()
	{
		POINT pos;
		GetCursorPos(&pos);
		return { pos.x, pos.y };
	}

	bool Mouse::IsLeftMouseDown()
	{
		return (GetKeyState(VK_LBUTTON) & 0x8000) != 0;
	}

	bool Mouse::IsRightMouseDown()
	{
		return (GetKeyState(VK_RBUTTON) & 0x8000) != 0;
	}

	bool Mouse::IsMiddleMouseDown()
	{
		return (GetKeyState(VK_MBUTTON) & 0x8000) != 0;
	}

}

File: WrecklessScript.Core/Input/Mouse.cs
using System;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

namespace WrecklessScript.Core
{
    namespace Input
    {
		[StructLayout(LayoutKind.Sequential)]
		public struct MouseCoords
		{
			public int X;
			public int Y;
		};

		public static class Mouse
        {
			[MethodImpl(MethodImplOptions.InternalCall)]
			public static extern void SetPosition(int x, int y);
			[MethodImpl(MethodImplOptions.InternalCall)]
			public static extern int GetPositionX();
			[MethodImpl(MethodImplOptions.InternalCall)]
			public static extern int GetPositionY();
			[MethodImpl(MethodImplOptions.InternalCall)]
			public static extern MouseCoords GetPosition();
			[MethodImpl(MethodImplOptions.InternalCall)]
			public static extern bool IsLeftMouseDown();
			[MethodImpl(MethodImplOptions.InternalCall)]
			public static extern bool IsRightMouseDown();
		}
    }
}
